middleware/ChaosHandler.md

```diff
@@ -1,30 +1,38 @@
 # Chaos Handler
-A piece of middleware that helps to mock different kinds of responses.
+
+A piece of middleware that enables developers to inject server failures that are difficult to trigger. The goal of this Chaos Handler is to allow a developer to test their application code with random errors injected into their application so that they can see whether the application handles server generated failures. Additionally, a developer can explicitly control the generated failures to assist with reproducing the failure and aid in fixing the application behaviour under failure conditions. 
 
 ## Objectives
-Provide a component, so that application developers can test their application with different kinds of responses (different response codes).
+
+Provide a configurable testing handler component that, by default, generates random error responses, and can be configured to generate tester specified errors.
 
 ## Requirements
+
+* MUST support request/response pass-through for non-intercepted responses.
+* The rate of error responses returned MUST be configurable with a value that represents the percentage of responses intercepted and replaced with a random error response. 
+* MUST NOT be a default handler in the midleware chain.
+* It SHOULD be easily added to the default middleware chain without affecting the expected middleware functionality.
+* It SHOULD be added as the handler immediately before the final handler.
+* The default Chaos Handler SHOULD be in random mode by default
 * Should be included in middleware using Custom Middleware Chain.
-* Should support Manual Mode of operation, where application developer can specify what response code they want for which API request.
+* Should support Manual Mode of operation, where application developer can specify what responses they want for a provided request.
```

Comment:
I feel that's a broad statement that would need a bit more precision on how that should work. But since it was already there I'm happy to leave that for another time.

Reply:
Naw, let's address it real quick.

Reply:
Nope, I take that back. Let's leave it. Thank you!